Help to reduce size of mysql client installation by adding an option to remove
unneeded stuff for most everyday use.

Signed-off-by: Fabien Marteau <fabien.marteau@armadeus.com>

Index: buildroot/package/database/mysql_client/Config.in
===================================================================
--- buildroot.orig/package/database/mysql_client/Config.in	2008-11-03 09:15:31.000000000 +0100
+++ buildroot/package/database/mysql_client/Config.in	2010-09-08 14:05:00.000000000 +0200
@@ -2,3 +2,21 @@
 	bool "MySQL 5.1 client"
 	help
 	  MySQL client
+
+if BR2_PACKAGE_MYSQL_CLIENT
+config BR2_PACKAGE_MYSQL_CLIENT_SHRINK
+	bool "MySQL client shrink"
+	help
+	  Reduce the size of MySQL package by suppressing:
+	  usr/mysql-test/
+	  usr/sql-bench/
+	  usr/bin/mysql_client_test
+	  usr/bin/mysql_waitpid
+	  usr/bin/myisampack
+	  usr/bin/mysqldump
+	  usr/bin/mysqltest
+	  usr/bin/myisamchk
+	  usr/bin/myisam_ftdump
+	  usr/bin/myisamlog
+	  usr/bin/my_print_defaults
+endif
Index: buildroot/package/database/mysql_client/mysql_client.mk
===================================================================
--- buildroot.orig/package/database/mysql_client/mysql_client.mk	2008-11-03 09:15:31.000000000 +0100
+++ buildroot/package/database/mysql_client/mysql_client.mk	2010-09-08 15:15:51.000000000 +0200
@@ -28,3 +28,20 @@
 	$(ENABLE_DEBUG)
 
 $(eval $(call AUTOTARGETS,package/database,mysql_client))
+
+$(MYSQL_CLIENT_HOOK_POST_INSTALL):
+	cd $(TARGET_DIR)/usr/lib/;cp mysql/libmysqlclient.so.16.0.0 libmysqlclient.so.16
+ifeq ($(BR2_PACKAGE_MYSQL_CLIENT_SHRINK),y)
+	rm -rf $(TARGET_DIR)/usr/mysql-test/
+	rm -rf $(TARGET_DIR)/usr/sql-bench/
+	rm -rf $(TARGET_DIR)/usr/bin/mysql_client_test
+	rm -rf $(TARGET_DIR)/usr/bin/mysql_waitpid
+	rm -rf $(TARGET_DIR)/usr/bin/myisampack
+	rm -rf $(TARGET_DIR)/usr/bin/mysqldump
+	rm -rf $(TARGET_DIR)/usr/bin/mysqltest
+	rm -rf $(TARGET_DIR)/usr/bin/myisamchk
+	rm -rf $(TARGET_DIR)/usr/bin/myisam_ftdump
+	rm -rf $(TARGET_DIR)/usr/bin/myisamlog
+	rm -rf $(TARGET_DIR)/usr/bin/my_print_defaults
+endif
+	touch $@
